netket/models/equivariant.py

```diff
@@ -0,0 +1,141 @@
+
+    symmetries: Callable[[], Array]
+    """permutations specifying symmetry group"""
+    group_algebra: Tuple
+    """Matrix specifying algebra of symmetry group given by SymmGroup"""
+    layers: int
+    """Number of layers (not including sum layer over output)"""
+    features: Union[Tuple, int]
+    """Number of features in each layer starting from the input. If
+    a single number is given, all layers will have the same number
+    of features"""
+    dtype: Any = np.float64
+    """The dtype of the weights."""
+    activation: Any = nknn.relu
+    """The nonlinear activation function between hidden layers."""
+    output_activation: Any = nknn.logcosh
+    """The nonlinear activation before the output."""
+    use_bias: bool = True
+    """if True uses a bias in all layers."""
+    precision: Any = None
+    """numerical precision of the computation see `jax.lax.Precision`for details."""
+    kernel_init: NNInitFunc = variance_scaling(1.0, "fan_in", "normal")
+    """Initializer for the Dense layer matrix."""
+    bias_init: NNInitFunc = zeros
+    """Initializer for the hidden bias."""
+
+    def setup(self):
+
+        self.n_symm, _ = self.symmetries().shape
+
+        if isinstance(self.features, int):
+            feature_dim = [self.features for layer in range(self.layers)]
+        else:
+            if not len(self.features) == self.layers:
+                raise ValueError(
+                    """Length of vector specifying feature dimensions must be the same as the number of layers"""
+                )
+            else:
+                feature_dim = tuple(self.features)
+
+        self.dense_symm = nknn.DenseSymm(
+            symmetries=self.symmetries,
+            features=feature_dim[0],
+            dtype=self.dtype,
+            use_bias=self.use_bias,
+            kernel_init=self.kernel_init,
+            bias_init=self.bias_init,
+            precision=self.precision,
+        )
+
+        self.equivariant_layers = [
+            nknn.DenseEquivariant(
+                group_algebra=self.group_algebra,
+                in_features=feature_dim[layer],
+                out_features=feature_dim[layer + 1],
+                use_bias=self.use_bias,
+                dtype=self.dtype,
+                precision=self.precision,
+                kernel_init=self.kernel_init,
+                bias_init=self.bias_init,
+            )
+            for layer in range(self.layers - 1)
+        ]
+
+    @nn.compact
+    def __call__(self, x_in):
+        x = self.dense_symm(x_in)
+        for layer in range(self.layers - 1):
+            x = self.activation(x)
+            x = self.equivariant_layers[layer](x)
+
+        x = self.output_activation(x)
+        x = jnp.sum(x, axis=-1)
+
+        return x
+
+
+def create_GCNN(
+    symmetries: Union[AbstractGraph, Array],
+    *args,
+    **kwargs,
+):
+    """
+    Constructor for GCNN
+    """
+
+    if isinstance(symmetries, AbstractGraph):
```

Comment:
We opted for not taking in a graph when symmetries are required. It's either a matrix or symmgroup. So remove this if branch

Reply:
Yes, I read the PR and I've implemented the changes.  Now the constructors are gone and GCNN excepts a SymmGroup or a HashableArray. If it's an array, you also need to specify group_algebra

Reply:
Yeah sorry, all is fine.
I guess i did not refresh the page... my bad

Reply:
No problem, just didn't want to waste your time looking through my code when I was fixing things already :)